rpc/client.go

```diff
@@ -106,15 +107,38 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 		Closed:       make(chan struct{}),
 		clock:        context.localClock,
 		remoteClocks: context.RemoteClocks,
-		cached:       !context.DisableCache,
 	}
-	if !context.DisableCache {
-		clients[c.Addr().String()] = c
+	clients[c.Addr().String()] = c
+
+	tlsConfig, tlsErr := context.GetClientTLSConfig()
+
+	connect := func() error {
```

Comment:
This function appears to only be called once, so it could just be inlined into the RunWorker closure. But I agree with Marc that we should probably either add an error return to NewClient and fail immediately if we can't load the TLS configs, or delay the call to GetClientTLSConfig until we're actually trying to connect. I prefer the latter since everything's cached; we generally don't have great error-handling options at the point where we're calling NewClient.